libs/agno/agno/knowledge/agent.py

```diff
@@ -123,8 +162,7 @@ def load(
 
             # Upsert documents if upsert is True and vector db supports upsert
             if upsert and self.vector_db.upsert_available():
-                for doc in document_list:
-                    self.vector_db.upsert(documents=[doc], filters=doc.meta_data)
+                self.vector_db.upsert(documents=documents_to_load, filters=doc.meta_data)
```

Comment:
`documents_to_load` is only defined after this line (line 171), this would not work here right